### PR TITLE
#509 Added home page

### PR DESCRIPTION
**Pull request by szczepanbarszczowski**

### CODE REVIEW

**Comment by ArekRado** [comment]
build doesn't work

File: src/pages/home/Home.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import React from 'react'
```
Comment (ArekRado) [inline src/pages/home/Home.js]: do not put home folder in pages. It will create new route `/home` move these files to folder `/routes/home`

**Comment by ArekRado** [comment]
some components should be moved to `/routes/home/components` because they are used only in home route and it's unclear to store them in `/components` folder
eg:
- WhatWeDo
- ContactSection
- Clients

**Comment by ArekRado** [comment]
images are pretty huge, do not use images with `@3 @2` prefixes + use **twice** tinyjpg.com

**Comment by ArekRado** [comment]
carousel doesn't work

image

**Comment by ArekRado** [comment]
fix propTypes errors

File: src/pages/home/Home.js (hunk 1 above)
Comment (szczepanbarszczowski) [reply src/pages/home/Home.js]: +1

File: src/pages/thank-you/ThankYou.module.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+.desc {
+  @include font(base, $color-gray, $align: center);
+  @include my(2);
+}
+
+.socialMedia {
+  justify-content: center;
+
+  @include media-breakpoint-down(sm) {
+    div {
```
Comment (ArekRado) [inline src/pages/thank-you/ThankYou.module.scss]: remove tag selector and add:

```scss
@include flex(column, center, center);
```

File: src/routes/Home/Home.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import { WhatWeDo } from './components/WhatWeDo/WhatWeDo'
+import { ContactSection } from './components/ContactSection/ContactSection'
+import { ClutchSection } from './components/ClutchSection/ClutchSection'
+import { Clients } from './components/Clients/Clients'
+import { Hr } from 'components/Hr/Hr'
+import { contactUrl, estimateUrl } from 'utils/url'
+
+import c from './Home.module.scss'
+
+const HomeComponent = pure(({ lessThanMedium }) => (
```
Comment (ArekRado) [inline src/routes/Home/Home.js]: connect is pure by default, you can remove pure()

File: src/routes/Home/Home.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import pure from 'recompose/pure'
+import PropTypes from 'prop-types'
+import cn from 'classnames'
+import { connect } from 'react-redux'
+
+import { ArrowButton } from 'components/Controls/ArrowButton'
+import { WhatWeDo } from './components/WhatWeDo/WhatWeDo'
```
Comment (ArekRado) [inline src/routes/Home/Home.js]: use index.js for shorter imports

File: src/routes/Home/Home.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import { ContactSection } from './components/ContactSection/ContactSection'
+import { ClutchSection } from './components/ClutchSection/ClutchSection'
+import { Clients } from './components/Clients/Clients'
+import { Hr } from 'components/Hr/Hr'
```
Comment (ArekRado) [inline src/routes/Home/Home.js]: same as above and above :D

File: src/routes/Home/Home.module.scss [hunk 6]
```diff
@@ -0,0 +1,96 @@
+@import '~styles/base';
+
+.wrapper {
+  @include relative;
+  height: 100%;
```
Comment (ArekRado) [inline src/routes/Home/Home.module.scss]: size mixin

File: src/routes/Home/Home.module.scss [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  width: 100%;
+  overflow: hidden;
+
+  .headingSection {
+    @include py(5);
+    width: 100%;
```
Comment (ArekRado) [inline src/routes/Home/Home.module.scss]: size mixin

File: src/routes/Home/Home.module.scss [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  .getInTouchBtn {
+    @include my(1);
+    align-self: flex-end;
+  }
+}
+
+.clutchBg {
+  @include absolute(bottom 0 left 0 right 0);
+
+  height: 769px;
```
Comment (ArekRado) [inline src/routes/Home/Home.module.scss]: size mixin

File: src/routes/Home/Home.module.scss [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  width: 100%;
+}
+
+.carousel {
+  height: 480px;
```
Comment (ArekRado) [inline src/routes/Home/Home.module.scss]: size mixin

File: src/routes/Home/components/Clients/Clients.module.scss [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    width: 700px;
+
+    @include media-breakpoint-down(sm) {
+      @include p(0);
+      @include pt(1);
+      width: 100%;
+    }
+
+    .nike {
+      max-height: 33px;
```
Comment (ArekRado) [inline src/routes/Home/components/Clients/Clients.module.scss]: why not  svg-icon mixin?

File: src/routes/Home/components/ContactSection/ContactSection.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+  <div className={c.wrapper}>
+    <h1 className={c.title}>Let us design your idea</h1>
+
+    <div className={c.formWrap}>
+      <ContactForm className={c.form} />
+    </div>
+  </div>
+)
+
+export default ContactSection
```
Comment (ArekRado) [inline src/routes/Home/components/ContactSection/ContactSection.js]: no default :D

File: src/routes/Home/components/ContactSection/ContactSection.module.scss [hunk 12]
```diff
@@ -0,0 +1,48 @@
+@import '~styles/base';
+
+.wrapper {
+  padding-top: 113px;
+  padding-bottom: 200px;
+
+  @include media-breakpoint-down(sm) {
+    padding-top: 120px;
```
Comment (ArekRado) [inline src/routes/Home/components/ContactSection/ContactSection.module.scss]: py(7)

File: src/routes/Home/components/ContactSection/components/ContactForm/ContactForm.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+export default connect(
+  state => ({
+    isSubmitting: state.estimate.submitting,
+  }),
+  { onSubmit: estimateRequest }
+)(
+  reduxForm({
+    form: 'estimate',
+    pure: true,
```
Comment (ArekRado) [inline src/routes/Home/components/ContactSection/components/ContactForm/ContactForm.js]: connect is pure by default, are we really need pure form?

File: src/routes/Home/components/ContactSection/components/ContactForm/ContactForm.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+      ]}
+    />
+
+    <ArrowButton submitting={isSubmitting} type="submit" className={c.btn}>
+      Send the message
+    </ArrowButton>
+  </form>
+)
+
+export default connect(
```
Comment (ArekRado) [inline src/routes/Home/components/ContactSection/components/ContactForm/ContactForm.js]: no-default

File: src/routes/Home/components/WhatWeDo/WhatWeDo.js [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+          text="We raise efficiency for well established brands"
+          firstWord="Big"
+          secondWord="Brands"
+          buttonText="See Dior Case Study"
+          linkUrl={contactUrl()}
+        />
+      </div>
+
+      <div className={cn(c.secondCol, c.col)}>
+        <div>
```
Comment (ArekRado) [inline src/routes/Home/components/WhatWeDo/WhatWeDo.js]: no-empty-divs :V

File: src/routes/Home/components/WhatWeDo/WhatWeDo.js [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+          <img src={diorSection} alt="Dior" className={c.img} />
+        </div>
+      </div>
+    </div>
+
+    <div className={cn(c.secondSection, c.section)}>
+      <div className={cn(c.firstCol, c.col)}>
+        <div>
```
Comment (ArekRado) [inline src/routes/Home/components/WhatWeDo/WhatWeDo.js]: same as above

File: src/routes/Home/components/WhatWeDo/WhatWeDo.js [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+          text="We empower amazing startup ideas"
+          firstWord="New"
+          secondWord="Startups"
+          buttonText="See Flymble Case Study"
+          linkUrl={contactUrl()}
+        />
+      </div>
+
+      <div className={cn(c.secondCol, c.col)}>
+        <div>
```
Comment (ArekRado) [inline src/routes/Home/components/WhatWeDo/WhatWeDo.js]: same as aboce

File: src/routes/Home/components/WhatWeDo/WhatWeDo.module.scss [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+    .secondCol {
+      background-color: #f6f6f6;
+      @include media-breakpoint-down(sm) {
+        order: 1;
+      }
+    }
+  }
+
+  .secondSection {
+    margin-top: 200px;
```
Comment (ArekRado) [inline src/routes/Home/components/WhatWeDo/WhatWeDo.module.scss]: mt(7)

File: src/routes/Home/components/WhatWeDo/WhatWeDo.module.scss [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+      padding-left: 140px;
+      @include media-breakpoint-down(sm) {
+        margin-top: 20px;
+        padding-left: 0;
+      }
+    }
+  }
+
+  .thirdSection {
+    margin-top: 200px;
```
Comment (ArekRado) [inline src/routes/Home/components/WhatWeDo/WhatWeDo.module.scss]: mt(7)

File: src/styles/_variables.scss [hunk 20]
```diff
@@ -127,7 +145,7 @@ $spacing-sizes: (
   2: 30px,
   3: 50px,
   4: 75px,
-  5: 120px,
+  5: 90px,
```
Comment (ArekRado) [inline src/styles/_variables.scss]: you really don't want to do it ;)

File: src/styles/_variables.scss [hunk 21]
```diff
@@ -159,21 +177,24 @@ $typography: (
...
       font-size: 48px,
       font-weight: 300,
-      line-height: 52px,
+      line-height: 72px,
     ),
-    sm: (
-      font-size: 24px,
-      line-height: 52px,
+  ),
+  h1-bold: (
```
Comment (ArekRado) [inline src/styles/_variables.scss]: I think it was mistake to create h1, h1-light, h1-bold.... if we want to change single property we should add eg font-family

File: src/styles/_variables.scss [hunk 22]
```diff
@@ -215,14 +244,14 @@ $typography: (
...
-      font-size: 20px,
+      font-size: 24px,
       line-height: 28px,
     ),
   ),
   h5: (
     base: (
       font-family: $font-family-base,
-      font-size: 18px,
+      font-size: 14px,
```
Comment (ArekRado) [inline src/styles/_variables.scss]: it' could break another pages, did you check it?

File: src/styles/_variables.scss [hunk 23]
```diff
@@ -315,6 +349,14 @@ $typography: (
       font-weight: 300,
     ),
   ),
+  light: (
```
Comment (ArekRado) [inline src/styles/_variables.scss]: light what?

**Comment by ArekRado** [comment]
images are still to big, use images with smaller dimensions **(not @2x @3x)**

File: src/components/Hr/Hr.js [hunk 24]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import pure from 'recompose/pure'
+
+import c from './Hr.module.scss'
+
+export const Hr = pure(({ text }) => (
```
[the remark this answers is not in the document]
Comment (szczepanbarszczowski) [reply src/components/Hr/Hr.js]: It's always text.

File: src/components/Controls/ArrowButton.js [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+    ) : (
+      <button className={cn(c[theme], className, c.arrowBtn)} disabled={submitting}>
+        <span className={c.arrow}>{submitting ? <Loader className={c.loader} /> : <ArrowRight />}</span>
+        <div className={c.content}>{children}</div>
+      </button>
+    )
+)
+
+ArrowButton.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
```
[the remark this answers is not in the document]
Comment (szczepanbarszczowski) [reply src/components/Controls/ArrowButton.js]: +1

File: src/components/Controls/ArrowButton.js [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+import Loader from 'components/Loader/Loader'
+import { Link } from './Link'
+
+import c from './ArrowButton.module.scss'
+
+export const ArrowButton = pure(
+  ({ children, className, theme, submitting, to }) =>
+    to ? (
+      <Link className={cn(c[theme], className, c.arrowBtn)} to={to}>
+        <span className={c.arrow}>{submitting ? <Loader className={c.loader} /> : <ArrowRight />}</span>
```
[the remark this answers is not in the document]
Comment (szczepanbarszczowski) [reply src/components/Controls/ArrowButton.js]: +1

File: src/components/Controls/ArrowButton.js [hunk 27]
```diff
@@ -0,0 +1,37 @@
...
+        <div className={c.content}>{children}</div>
+      </button>
+    )
+)
+
+ArrowButton.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
+  className: PropTypes.string,
+  theme: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (szczepanbarszczowski) [reply src/components/Controls/ArrowButton.js]: Why?

File: src/components/Loader/Loader.js [hunk 28]
```diff
@@ -4,7 +4,7 @@ import c from './Loader.module.scss'
 import { Loader as ReactLoader } from 'react-loaders'
 
 const Loader = ({ className, theme = 'default', ...props }) => {
-  return <ReactLoader className={cn(c.loader, c[theme], className)} {...props} type="line-scale" />
+  return <ReactLoader className={cn(c.loader, c[theme], className)} {...props} type="ball-beat" />
 }
 
 export default Loader
```
[the remark this answers is not in the document]
Comment (szczepanbarszczowski) [reply src/components/Loader/Loader.js]: +1

File: src/routes/Home/components/WhatWeDo/WhatWeDo.module.scss (hunk 19 above)
Comment (szczepanbarszczowski) [reply src/routes/Home/components/WhatWeDo/WhatWeDo.module.scss]: +1

File: src/styles/_variables.scss (hunk 20 above)
Comment (szczepanbarszczowski) [reply src/styles/_variables.scss]: +1

File: src/styles/_variables.scss (hunk 21 above)
Comment (szczepanbarszczowski) [reply src/styles/_variables.scss]: It's not single property. We have font-family, line-height and font-weight which differ from h1.

File: src/styles/_variables.scss (hunk 22 above)
Comment (szczepanbarszczowski) [reply src/styles/_variables.scss]: Sure, will change to avoid conflicts with your code.

**Comment by szczepanbarszczowski** [comment]
I'll add carousel in separate branch.

**Pull request merged**